sqlmesh/core/test/definition.py

```diff
@@ -763,3 +765,32 @@ def _normalize_df_value(value: t.Any) -> t.Any:
             return {k: _normalize_df_value(v) for k, v in zip(value["key"], value["value"])}
         return {k: _normalize_df_value(v) for k, v in value.items()}
     return value
+
+
+def _load_csv(path: Path | str | None, rows: str | None) -> t.List | None:
+    """Handler to load rows from csv file or string."""
+    return pd.read_csv(path or StringIO(rows)).to_dict(orient="records")
+
+
+def _load_yaml(path: Path | str | None, rows: t.List | None) -> t.List | None:
+    """Handler to load rows from yaml file or list."""
+    input_rows = yaml_load(Path(path)) if path else rows
+    return input_rows.get("rows") if isinstance(input_rows, dict) else input_rows
+
+
+FORMAT_HANDLERS = {"csv": _load_csv, "yaml": _load_yaml}
+
+
+def _load_rows(values: t.Dict) -> t.List | None:
+    """Load data in rows from external file or inline declared string."""
+    rows = values.get("rows")
+    format = values.get("format") or Path(values.get("path", "")).suffix[1:]
```

Comment:
Ok, since the inference will be removed for now, when inline rows are provided it will be handled as YAML. 

But when there is a path (without format) should it handle it as YAML and try and load the file or should this not be handled and enforce that the user should have defined the test correctly?

Reply:
We could try to load the YAML, @tobymao what's your preference here?